core/main.go

```diff
@@ -119,14 +122,37 @@ func main() {
 		logrus.WithField("module", "main").Fatalln(err)
 	}
 
+	genesis, err := cosmos.LoadGenesis(cfg.Tendermint.GenesisFile())
+	// generate genesis if it doesn't exist
+	if os.IsNotExist(err) {
+		logrus.WithField("module", "main").Warn("Genesis file not found. Will generate an unsecured developer one.")
+		validator, err := cosmos.NewGenesisValidator(kb,
+			cfg.DevGenesis.AccountName,
+			cfg.DevGenesis.AccountPassword,
+			cfg.Tendermint.PrivValidatorKeyFile(),
+			cfg.Tendermint.PrivValidatorStateFile(),
+			cfg.Tendermint.NodeKeyFile(),
+		)
+		if err != nil {
+			logrus.WithField("module", "main").Fatalln(err)
+		}
+		genesis, err = cosmos.GenGenesis(app, kb, cfg.DevGenesis.ChainID, cfg.Tendermint.GenesisFile(), []cosmos.GenesisValidator{validator})
+		if err != nil {
+			logrus.WithField("module", "main").Fatalln(err)
+		}
+		logrus.WithFields(validator.Map()).Infoln("Validator")
+	} else if err != nil {
+		logrus.WithField("module", "main").Fatalln(err)
+	}
+
```

Comment:
The problem if config is handling the loading or generation of the genesis is that config will have a huge dependency to cosmos and tendermint. Also, to generate the generate the genesis, cosmos app, keybase are required.

Reply:
I moved this logic to a specific function in `main.go` to ease a bit the main function

Reply:
I see. I thought that it will need only the tendermint and cosmos package (not the one created by us), but for generate genesis validator, it won't need any extra dependencies. Only to generate the app state. 

So I'm ok to keep validator generation in cosmos (although I would put that in config :P),
but what do you think about moving that function cosmos package itself. 

- I want to keep main.go as small as possible. 
- This logic doesn't need any data from main except config.
- in cosmos package that function could be tested easily.